azurerm/internal/services/web/registration.go

```diff
@@ -27,6 +27,7 @@ func (r Registration) SupportedDataSources() map[string]*schema.Resource {
 		"azurerm_app_service_certificate":       dataSourceAppServiceCertificate(),
 		"azurerm_app_service_plan":              dataSourceAppServicePlan(),
 		"azurerm_function_app":                  dataSourceArmFunctionApp(),
+		"azurerm_function_app_host_keys":        dataSourceArmFunctionAppHostKeys(),
```

Comment:
Can we rename this resource as "azurerm_app_service_host_key", since:

1. Making it non-plural as it is only for a certain host
2. Name it as app service instead of function app (though it should works for both as both are using the same API at the end) to make it align with other resource naming conventions (e.g. `azurerm_app_service_slot_virtual_network_swift_connection`)

So you might also need to update the document and rename files accordingly.

Reply:
this makes more sense as `keys`, since it's returning (presumably) the primary and secondary keys here - so this is fine as plural